astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,397 @@
+                'In Astropy, "GPS" is a time from epoch format which runs '
+                'synchronously with TAI, as the scale GPS ≈ TAI − 19 s. '
+                'Hence, this format will be used.'.format(column_info),
+                AstropyUserWarning)
+            column_info['FORMAT'] = 'gps'
+            column_info['TCTYP'] = 'tai'
+            return True
+
+        if scale == 'LOCAL':
+            warnings.warn(
+                'Table column {} has a FITS recognized time scale value "LOCAL". '
+                'However, the standard states that "LOCAL" should be tied to one '
+                'of the existing scales because it is intrinsically unreliable '
+                'and/or ill-defined. Astropy will thus use the global time scale '
+                '(TIMESYS) as the default.'. format(column_info),
+                AstropyUserWarning)
+            column_info['TCTYP'] = global_info['TIMESYS']
+            column_info['FORMAT'] = global_info.get('FORMAT', None)
+
+        # Non-linear coordinate types have "4-3" form and are not time coordinates
+        if TCTYP_RE_TYPE.match(scale[:5]) and TCTYP_RE_ALGO.match(scale[5:]):
+            return False
+
+        # Coordinate type is either an unrecognized local time scale
+        # or a linear coordinate type
+        return False
+
+    if global_info['TIMESYS'] is None:
+        return False
+
+    if (unit is not None and unit in FITS_TIME_UNIT) or pos is not None:
+        column_info['TCTYP'] = global_info['TIMESYS']
+        return True
+
+    return False
+
+
+def _convert_global_time(table, global_info):
+    """
+    Convert the table metadata for time informational keywords
+    to astropy Time.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        The table whose time metadata is to be converted.
+    global_info : dict
+        Global time reference frame information.
+    """
+
+    # Read in Global Informational keywords as Time
+    for key, value in global_info.items():
+        # FITS uses a subset of ISO-8601 for DATE-xxx
+        if key.startswith('DATE'):
+            if key not in table.meta:
+                scale = 'utc' if key == 'DATE' else global_info['TIMESYS']
+                try:
+                    precision = len(value.split('.')[-1]) if '.' in value else 0
+                    value = Time(value, format='fits', scale=scale,
+                                 precision=precision)
+                except ValueError:
+                    pass
+                table.meta[key] = value
+
+        # MJD-xxx in MJD according to TIMESYS
+        elif key.startswith('MJD-'):
+            if key not in table.meta:
+                try:
+                    value = Time(value, format='mjd',
+                                 scale=global_info['TIMESYS'])
+                except ValueError:
+                    pass
+                table.meta[key] = value
+
+
+def _convert_time_columns(col, column_info):
```

Comment:
Note that this name should be `_convert_time_column` now - i.e., drop the last `s` -- at least that is what you call it below.

p.s. I strongly recommend you use an editor that checks for these kinds of mistakes automatically (like emacs, properly set up).